src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -289,10 +289,16 @@ protected function createNotFoundException(string $message = 'Not Found', \Excep
      *
      *     throw $this->createAccessDeniedException('Unable to access this page!');
      *
+     * @throws \LogicException If SecurityBundle is not available
+     *
      * @final since version 3.4
      */
     protected function createAccessDeniedException(string $message = 'Access Denied.', \Exception $previous = null): AccessDeniedException
     {
+        if (!class_exists(AccessDeniedException::class)) {
+            throw new \LogicException('You can not use the "createAccessDeniedException" method if the SecurityBundle is not registered in your application.');
```

Comment:
To be consistent with the error messages displayed in other parts of the framework, we should change this message to:

```php
throw new \LogicException('The SecurityBundle is not registered in your application. Try running "composer require security"');
```

Reply:
@javiereguiluz : the message you suggest is only used in three methods in the file: `getDoctrine()`, `getUser()`, `isGranted()`, while the suggested choice is used in four methods: `addFlash()`, `renderView()`, `render()`, and `stream()`, hence the current wording choice, favoring the longer format for internal consistency of this file. 

Is there a trend either way in recent Symfony changes, to ensure the consistency goes the proper way at a larger scale ?

Reply:
You are right!! However, I was mostly referring to the *"try running composer require ..."* part of the error message. It's an absolute must for newer Symfony apps using Flex. We do that in other error messages:

```php
throw new LogicException('CSRF support cannot be enabled as the Security CSRF component is not installed. Try running "composer require security-csrf".');

throw new \LogicException(sprintf('The type="expression" attribute cannot be used without the ExpressionLanguage component. Try running "composer require symfony/expression-language".'));
```

And that trend will continue because of PRs like this: https://github.com/symfony/symfony/pull/25601

Reply:
Oh, I see. There is not such wording in the current version, but it does fit with the push for Flex. Done.